File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        if (is_sorted(nums.begin(), nums.end()))
            return true;
        
        for (int i = 0, n = nums.size(); i < n; i++) {
            // first i elements at the back
            // final arr = [i + 1....n, 0...i]

            vector<int> tmp;
            // tmp.resize(n);
            tmp.insert(tmp.begin(), nums.begin() + i + 1, nums.end());
            tmp.insert(tmp.end(), nums.begin(), nums.begin() + i + 1);
            

            if (is_sorted(tmp.begin(), tmp.end()))  return true;
        }
        return false;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    private boolean isSorted(int[] nums) {
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] >= nums[i-1])
                continue;
            return false;
        }
        return true;
    }
    public boolean check(int[] nums) {
        if (isSorted(nums))
            return true;

        final int n = nums.length;

        for (int i = 0; i < n; i++) {
            // First i elements at the back
            // final arr = [i + 1, ... n, 0 ... i]
            int[] tmp = new int[n];
            System.arraycopy(nums, i + 1, tmp, 0, n - i - 1);
            System.arraycopy(nums, 0, tmp, n - i - 1, i + 1);

            if (isSorted(tmp))
                return true;
        }

        return false;

        // return false;
    }
}